### DISCUSSION TITLE
Being a good citizen: New ESMA_cmake include and External users

### DISCUSSION THREAD

**Discussion by mathomp4** [GEOS-ESM/MAPL#988] [General]
This is a question/discussion for @rmontuoro (and maybe @LiamBindle?). 

Soon-ish, I will be [including a new file, `esma_cpack.cmake`](https://github.com/GEOS-ESM/ESMA_cmake/pull/200) in our ESMA_cmake repo. Now, because of ecbuild, I can't just "hide" it inside of our `esma.cmake` omnibus file where it wouldn't matter. Rather, I'll have to add to the end of MAPL's `CMakeLists.txt` file  a la:
```cmake
...
# Piggyback that file into install
install(
   FILES ${PROJECT_BINARY_DIR}/.gitignore
   DESTINATION ${CMAKE_INSTALL_PREFIX}
   )

ecbuild_install_project (NAME MAPL)

# CPack
include(esma_cpack OPTIONAL)
```
Where those last two lines are the important ones. Now, my hope is that maybe the `OPTIONAL` is enough to get by, but there is stuff in CMakeLists that @rmontuoro added like:
```cmake
...
# build as standalone project
if (NOT COMMAND esma)
  include (esma OPTIONAL)
endif ()

if (NOT COMMAND esma)

  if (SKIP_MEPO)
    message (FATAL_ERROR "ESMA not found")
  else ()
...
```

Now, I'll work with @weiyuan-jiang to try it out with UFS first, but I thought I'd lean on @rmontuoro's experience and knowledge to see if you can think of more CMake protection that might be needed?

**Comment by LiamBindle** [comment]
I think we should be okay on our end. We include ESMA_cmake in GCHP so we'll get that new module when we update ESMA_cmake (unless I've misunerstanding something here). Thanks for the ping.
